src/System.Private.CoreLib/src/System/Utf8String.cs

```diff
@@ -0,0 +1,252 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Text;
+using Internal.Runtime.CompilerServices;
+
+namespace System
+{
+    /// <summary>
+    /// Represents an immutable string of UTF-8 code units.
+    /// </summary>
+    public sealed partial class Utf8String : IEquatable<Utf8String>
+    {
+        /*
+         * STATIC FIELDS
+         */
+
+        public static readonly Utf8String Empty = FastAllocate(0);
+
+        /*
+         * INSTANCE FIELDS
+         * Do not reorder these fields. They must match the layout of Utf8StringObject in object.h.
+         */
+
+        private readonly int _length;
+        private readonly byte _firstByte;
+
+        /*
+         * OPERATORS
+         */
+
+        /// <summary>
+        /// Compares two <see cref="Utf8String"/> instances for equality using a <see cref="StringComparison.Ordinal"/> comparer.
+        /// </summary>
+        public static bool operator ==(Utf8String a, Utf8String b) => Equals(a, b);
```

Comment:
Same comment as https://github.com/dotnet/coreclr/pull/23209/files#r264965988.

I realize `String`'s `==` operator uses `a` and `b`, but usually for new types, the guidelines are followed. For example, `DateTime` uses `d1` and `d2` as the parameter names, but when `DateTimeOffset` was added, the design guidelines were followed and `left` and `right` were used.